lib/Conversion/VerifToSV/VerifToSV.cpp

```diff
@@ -51,6 +53,83 @@ struct PrintOpConversionPattern : public OpConversionPattern<PrintOp> {
   }
 };
 
+struct HasBeenResetConversion : public OpConversionPattern<HasBeenResetOp> {
+  using OpConversionPattern<HasBeenResetOp>::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(HasBeenResetOp op, OpAdaptor operands,
+                  ConversionPatternRewriter &rewriter) const override {
+    auto i1 = rewriter.getI1Type();
+    auto constOne = rewriter.create<hw::ConstantOp>(op.getLoc(), i1, 1);
+    auto constZero = rewriter.create<hw::ConstantOp>(op.getLoc(), i1, 0);
+    auto constX = rewriter.create<sv::ConstantXOp>(op.getLoc(), i1);
+
+    // Declare the register that will track the reset state.
+    auto reg = rewriter.create<sv::RegOp>(
+        op.getLoc(), i1, rewriter.getStringAttr("hasBeenResetReg"));
+
+    auto clock = operands.getClock();
+    auto reset = operands.getReset();
+
+    // Explicitly initialize the register in an `initial` block. In general, the
+    // register will come up as X, but this may be overridden by simulator
+    // configuration options.
+    //
+    // In case the reset is async, check if the reset is already active during
+    // the `initial` block and immediately set the register to 1. Otherwise
+    // initialize to X.
+    auto assignOne = [&] {
+      rewriter.create<sv::PAssignOp>(op.getLoc(), reg, constOne);
+    };
+    auto assignX = [&] {
+      rewriter.create<sv::PAssignOp>(op.getLoc(), reg, constX);
+    };
+
+    rewriter.create<sv::InitialOp>(op.getLoc(), [&] {
+      if (op.getAsync())
+        rewriter.create<sv::IfOp>(op.getLoc(), reset, assignOne, assignX);
+      else
+        assignX();
+    });
```

Comment:
Do we need to use blocking assignment in initial block?  I remember it could cause a weird behaivor when using non-locking assignment https://stackoverflow.com/questions/30125565/non-blocking-and-blocking-assignments-dont-work-as-expected.

Reply:
Hmmm yeah you're right, we probably want this to be blocking in the initial block. Good catch!

Reply:
Done